### PR TITLE
Update phone number validation to be Singapore-specific

### PR DESCRIPTION
**Pull request by itsme-zeix**

### CODE REVIEW

**Comment by itsme-zeix** [comment]
Do we want phone numbers to be necessarily Singapore specific? It's possible that we give a warning to the user instead, but parse validity based on other criterias.

**Comment by ZShunRen** [comment]
> Do we want phone numbers to be necessarily Singapore specific? It's possible that we give a warning to the user instead, but parse validity based on other criterias.

I think this is a fair point, we could change it to give a warning instead of outright rejection, as there are cases of customers who may reside in Singapore but primarily overseas.

File: src/test/java/seedu/address/model/person/PhoneTest.java [hunk 1]
```diff
@@ -24,26 +24,46 @@ public void isValidPhone() {
...
+
+        // invalid phone numbers (additional characters)
+        assertFalse(Phone.isValidPhone(" 93123456 ")); // leading and trailing spaces
+        assertFalse(Phone.isValidPhone("\t93123456\n")); // tabs and newlines
+        assertFalse(Phone.isValidPhone("+6593121234")); // contains international prefix
+        assertFalse(Phone.isValidPhone("９３１２１５３４")); // Unicode digits (full-width characters)
         assertFalse(Phone.isValidPhone("9011p041")); // alphabets within digits
+        assertFalse(Phone.isValidPhone("9312-1534")); // special character within digits
+        assertFalse(Phone.isValidPhone("9312@1534")); // special character within digits
         assertFalse(Phone.isValidPhone("9312 1534")); // spaces within digits
```
Comment (ZShunRen) [inline src/test/java/seedu/address/model/person/PhoneTest.java:44]: I like the comprehensiveness of these test cases!

**Review by ZShunRen — APPROVED** [review]
LGTM, just make sure that running ./gradlew test, the rest of the tests don't break. If all tests pass, feel free to merge.

**Pull request merged**